hazelcast/src/test/java/com/hazelcast/internal/management/ExecuteScriptRequestTest.java

```diff
@@ -0,0 +1,114 @@
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ExecuteScriptRequestTest extends HazelcastTestSupport {
+
+    private TestHazelcastInstanceFactory factory;
+    private HazelcastInstance hz;
+
+    private ManagementCenterService managementCenterService;
+    private Map<String, Object> bindings = new HashMap<String, Object>();
+
+    @Before
+    public void setUp() {
+        factory = createHazelcastInstanceFactory(2);
+        hz = factory.newHazelcastInstance();
+        factory.newHazelcastInstance();
+
+        managementCenterService = getNode(hz).getManagementCenterService();
+
+        bindings.put("key", "value");
+    }
+
+    @After
+    public void tearDown() {
+        factory.terminateAll();
+    }
+
+    @Test
+    public void testExecuteScriptRequest() throws Exception {
+        ExecuteScriptRequest request = new ExecuteScriptRequest("print('test');", "JavaScript", false, bindings);
+
+        JsonObject jsonObject = new JsonObject();
+        request.writeResponse(managementCenterService, jsonObject);
+
+        JsonObject result = (JsonObject) jsonObject.get("result");
+        String response = (String) request.readResponse(result);
+        assertEquals("", response.trim());
+    }
+
+    @Test
+    public void testExecuteScriptRequest_whenTargetAllMembers() throws Exception {
+        ExecuteScriptRequest request = new ExecuteScriptRequest("print('test');", "JavaScript", true, null);
+
+        JsonObject jsonObject = new JsonObject();
+        request.writeResponse(managementCenterService, jsonObject);
+
+        JsonObject result = (JsonObject) jsonObject.get("result");
+        String response = (String) request.readResponse(result);
+        assertEquals(format("error%nerror%n"), response);
+    }
+
+    @Test
+    public void testExecuteScriptRequest_whenTargetAllMembers_withTarget() throws Exception {
+        Address address = hz.getCluster().getLocalMember().getAddress();
+        Set<String> targets = Collections.singleton(address.getHost() + ":" + address.getPort());
+        ExecuteScriptRequest request = new ExecuteScriptRequest("print('test');", "JavaScript", targets, bindings);
+
+        JsonObject jsonObject = new JsonObject();
+        request.writeResponse(managementCenterService, jsonObject);
+
+        JsonObject result = (JsonObject) jsonObject.get("result");
+        String response = (String) request.readResponse(result);
+        assertEquals("error", response.trim());
```

Comment:
I have no idea why this returns `error`, since it prints `test` in the log without any visible complains.